### ISSUE TITLE
error 4: wl_surface@20 already has a buffer committed

### ISSUE THREAD

**Issue by aleck099** [glfw/glfw#1492]
### Running the following code on arch linux you will see an output:
xdg_wm_base@9: error 4: wl_surface@20 already has a buffer committed.
![Screenshot from 2019-06-02 07-46-49](https://user-images.githubusercontent.com/33549441/58754806-8d919f00-84c7-11e9-8720-dc39153fb566.png)

### Code:
<details>
<summary>code</summary>

```cpp
#include <glad/glad.h>
#include <GLFW/glfw3.h>
#include <iostream>
#include <cmath>
#include <ctime>

void FrameBufferSizeCallback(GLFWwindow *, int width, int height)
{
	glViewport(0, 0, width, height);
}

int main()
{
	glfwInit();
	glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
	glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
	// error occured
	glfwWindowHint(GLFW_VISIBLE, GLFW_FALSE);
	GLFWwindow *main_window = glfwCreateWindow(800, 600, "Window Renderer", nullptr, nullptr);
	if (main_window == nullptr)
	{
		std::cout << "Failed to create window!\n";
		glfwTerminate();
		return 1;
	}
	glfwMakeContextCurrent(main_window);
	if (!gladLoadGLLoader((GLADloadproc)glfwGetProcAddress))
	{
		std::cout << "Failed to init glad!\n";
		glfwTerminate();
		return 2;
	}
	// error occured
	glfwShowWindow(main_window);
	glfwSetFramebufferSizeCallback(main_window, FrameBufferSizeCallback);
	while (!glfwWindowShouldClose(main_window))
	{
		float red_bit = fabs(sin(clock()));
		glClearColor(red_bit, 0.0f, 0.0f, 1.0f);
		glClear(GL_COLOR_BUFFER_BIT);
		glfwSwapBuffers(main_window);
		glfwPollEvents();
	}
	glfwTerminate();
	return 0;
}
```
</details>

But if I remove `glfwShowWindow` and `glfwWindowHint(GLFW_VISIBLE, GLFW_FALSE)`, making the window show on creating, everything works well.

Why cannot I create a window without showing it at once?

My display server is Wayland 1.17.0-1. It works well on Xorg.
Kernel version: 5.1.2-arch1-1-ARCH
glfw-wayland 3.3
gcc 8.3.0
glad sources are provided below
[glad4.zip](https://github.com/glfw/glfw/files/3244303/glad4.zip)

### Running with WAYLAND_DEBUG=1 I got these outputs:
<details>
<summary>debug outputs</summary>

```
[2893185.186]  -> wl_display@1.get_registry(new id wl_registry@2)
[2893185.243]  -> wl_display@1.sync(new id wl_callback@3)
[2893185.378] wl_display@1.delete_id(3)
[2893185.395] wl_registry@2.global(1, "wl_drm", 2)
[2893185.408] wl_registry@2.global(2, "wl_compositor", 4)
[2893185.422]  -> wl_registry@2.bind(2, "wl_compositor", 3, new id [unknown]@4)
[2893185.440] wl_registry@2.global(3, "wl_shm", 1)
[2893185.452]  -> wl_registry@2.bind(3, "wl_shm", 1, new id [unknown]@5)
[2893185.469] wl_registry@2.global(4, "wl_output", 2)
[2893185.482]  -> wl_registry@2.bind(4, "wl_output", 2, new id [unknown]@6)
[2893185.498] wl_registry@2.global(5, "zxdg_output_manager_v1", 1)
[2893185.511] wl_registry@2.global(6, "wl_data_device_manager", 3)
[2893185.523]  -> wl_registry@2.bind(6, "wl_data_device_manager", 1, new id [unknown]@7)
[2893185.540] wl_registry@2.global(7, "gtk_primary_selection_device_manager", 1)
[2893185.552] wl_registry@2.global(8, "wl_subcompositor", 1)
[2893185.564]  -> wl_registry@2.bind(8, "wl_subcompositor", 1, new id [unknown]@8)
[2893185.581] wl_registry@2.global(9, "xdg_wm_base", 2)
[2893185.593]  -> wl_registry@2.bind(9, "xdg_wm_base", 1, new id [unknown]@9)
[2893185.610] wl_registry@2.global(10, "zxdg_shell_v6", 1)
[2893185.622] wl_registry@2.global(11, "wl_shell", 1)
[2893185.635]  -> wl_registry@2.bind(11, "wl_shell", 1, new id [unknown]@10)
[2893185.651] wl_registry@2.global(12, "gtk_shell1", 3)
[2893185.664] wl_registry@2.global(13, "wp_viewporter", 1)
[2893185.676]  -> wl_registry@2.bind(13, "wp_viewporter", 1, new id [unknown]@11)
[2893185.693] wl_registry@2.global(14, "zwp_pointer_gestures_v1", 1)
[2893185.706] wl_registry@2.global(15, "zwp_tablet_manager_v2", 1)
[2893185.718] wl_registry@2.global(16, "wl_seat", 5)
[2893185.730]  -> wl_registry@2.bind(16, "wl_seat", 4, new id [unknown]@12)
[2893185.747] wl_registry@2.global(17, "zwp_relative_pointer_manager_v1", 1)
[2893185.760]  -> wl_registry@2.bind(17, "zwp_relative_pointer_manager_v1", 1, new id [unknown]@13)
[2893185.776] wl_registry@2.global(18, "zwp_pointer_constraints_v1", 1)
[2893185.789]  -> wl_registry@2.bind(18, "zwp_pointer_constraints_v1", 1, new id [unknown]@14)
[2893185.806] wl_registry@2.global(19, "zxdg_exporter_v1", 1)
[2893185.818] wl_registry@2.global(20, "zxdg_importer_v1", 1)
[2893185.830] wl_registry@2.global(21, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[2893185.843] wl_registry@2.global(22, "zwp_text_input_manager_v3", 1)
[2893185.855] wl_registry@2.global(23, "gtk_text_input_manager", 1)
[2893185.867] wl_callback@3.done(9702)
[2893185.874]  -> wl_display@1.sync(new id wl_callback@3)
[2893186.079] wl_display@1.delete_id(3)
[2893186.093] wl_output@6.geometry(0, 0, 480, 270, 1, "LEN", "LEN LT2223wA", 0)
[2893186.123] wl_output@6.mode(3, 1920, 1080, 60000)
[2893186.139] wl_output@6.scale(1)
[2893186.145] wl_output@6.done()
[2893186.150] wl_seat@12.capabilities(3)
[2893186.157]  -> wl_seat@12.get_pointer(new id wl_pointer@15)
[2893186.165]  -> wl_seat@12.get_keyboard(new id wl_keyboard@16)
[2893186.173] wl_seat@12.name("seat0")
[2893186.180] wl_callback@3.done(9702)
[2893186.305]  -> wl_shm@5.create_pool(new id wl_shm_pool@3, fd 7, 4096)
[2893186.488]  -> wl_shm_pool@3.resize(12288)
[2893186.572]  -> wl_shm_pool@3.resize(28672)
[2893187.985]  -> wl_shm_pool@3.resize(61440)
[2893188.043]  -> wl_shm_pool@3.resize(126976)
[2893188.115]  -> wl_shm_pool@3.resize(258048)
[2893188.245]  -> wl_shm_pool@3.resize(520192)
[2893191.392]  -> wl_shm_pool@3.resize(1044480)
[2893193.063]  -> wl_shm_pool@3.resize(2093056)
[2893205.881]  -> wl_shm@5.create_pool(new id wl_shm_pool@17, fd 9, 16384)
[2893206.105]  -> wl_shm_pool@17.resize(49152)
[2893206.267]  -> wl_shm_pool@17.resize(114688)
[2893210.310]  -> wl_shm_pool@17.resize(245760)
[2893210.409]  -> wl_shm_pool@17.resize(507904)
[2893210.586]  -> wl_shm_pool@17.resize(1032192)
[2893210.924]  -> wl_shm_pool@17.resize(2080768)
[2893219.148]  -> wl_shm_pool@17.resize(4177920)
[2893225.590]  -> wl_shm_pool@17.resize(8372224)
[2893271.883]  -> wl_compositor@4.create_surface(new id wl_surface@18)
[2893271.908]  -> wl_data_device_manager@7.get_data_device(new id wl_data_device@19, wl_seat@12)
[2893272.785]  -> wl_compositor@4.create_surface(new id wl_surface@20)
[2893272.793]  -> wl_compositor@4.create_region(new id wl_region@21)
[2893272.799]  -> wl_region@21.add(0, 0, 800, 600)
[2893272.810]  -> wl_surface@20.set_opaque_region(wl_region@21)
[2893272.815]  -> wl_surface@20.commit()
[2893272.818]  -> wl_region@21.destroy()
[2893274.194]  -> wl_display@1.get_registry(new id wl_registry@22)
[2893274.205]  -> wl_display@1.sync(new id wl_callback@23)
[2893274.581] wl_display@1.delete_id(21)
[2893274.592] wl_display@1.delete_id(23)
[2893274.597] wl_registry@22.global(1, "wl_drm", 2)
[2893274.607]  -> wl_registry@22.bind(1, "wl_drm", 2, new id [unknown]@21)
[2893274.620] wl_registry@22.global(2, "wl_compositor", 4)
[2893274.629] wl_registry@22.global(3, "wl_shm", 1)
[2893274.637] wl_registry@22.global(4, "wl_output", 2)
[2893274.646] wl_registry@22.global(5, "zxdg_output_manager_v1", 1)
[2893274.655] wl_registry@22.global(6, "wl_data_device_manager", 3)
[2893274.664] wl_registry@22.global(7, "gtk_primary_selection_device_manager", 1)
[2893274.672] wl_registry@22.global(8, "wl_subcompositor", 1)
[2893274.681] wl_registry@22.global(9, "xdg_wm_base", 2)
[2893274.690] wl_registry@22.global(10, "zxdg_shell_v6", 1)
[2893274.698] wl_registry@22.global(11, "wl_shell", 1)
[2893274.707] wl_registry@22.global(12, "gtk_shell1", 3)
[2893274.716] wl_registry@22.global(13, "wp_viewporter", 1)
[2893274.725] wl_registry@22.global(14, "zwp_pointer_gestures_v1", 1)
[2893274.733] wl_registry@22.global(15, "zwp_tablet_manager_v2", 1)
[2893274.742] wl_registry@22.global(16, "wl_seat", 5)
[2893274.751] wl_registry@22.global(17, "zwp_relative_pointer_manager_v1", 1)
[2893274.760] wl_registry@22.global(18, "zwp_pointer_constraints_v1", 1)
[2893274.768] wl_registry@22.global(19, "zxdg_exporter_v1", 1)
[2893274.777] wl_registry@22.global(20, "zxdg_importer_v1", 1)
[2893274.786] wl_registry@22.global(21, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[2893274.794] wl_registry@22.global(22, "zwp_text_input_manager_v3", 1)
[2893274.803] wl_registry@22.global(23, "gtk_text_input_manager", 1)
[2893274.812] wl_callback@23.done(9703)
[2893274.817]  -> wl_display@1.sync(new id wl_callback@23)
[2893274.989] wl_display@1.delete_id(23)
[2893275.000] wl_drm@21.device("/dev/dri/card0")
[2893275.040]  -> wl_drm@21.authenticate(8)
[2893275.046] wl_drm@21.format(808669761)
[2893275.051] wl_drm@21.format(808669784)
[2893275.055] wl_drm@21.format(808665665)
[2893275.059] wl_drm@21.format(808665688)
[2893275.064] wl_drm@21.format(875713089)
[2893275.068] wl_drm@21.format(875713112)
[2893275.072] wl_drm@21.format(909199186)
[2893275.077] wl_drm@21.format(961959257)
[2893275.081] wl_drm@21.format(825316697)
[2893275.085] wl_drm@21.format(842093913)
[2893275.089] wl_drm@21.format(909202777)
[2893275.094] wl_drm@21.format(875713881)
[2893275.098] wl_drm@21.format(842094158)
[2893275.102] wl_drm@21.format(909203022)
[2893275.106] wl_drm@21.format(1448695129)
[2893275.111] wl_drm@21.capabilities(1)
[2893275.115] wl_callback@23.done(9703)
[2893275.120]  -> wl_display@1.sync(new id wl_callback@23)
[2893275.200] wl_display@1.delete_id(23)
[2893275.206] wl_drm@21.authenticated()
[2893275.209] wl_callback@23.done(9703)
[2893329.379]  -> wl_surface@20.frame(new id wl_callback@23)
[2893329.440]  -> wl_drm@21.create_prime_buffer(new id wl_buffer@24, fd 14, 800, 600, 875713089, 0, 3328, 0, 0, 0, 0)
[2893329.513]  -> wl_surface@20.attach(wl_buffer@24, 0, 0)
[2893329.523]  -> wl_surface@20.damage(0, 0, 2147483647, 2147483647)
[2893331.876]  -> wl_surface@20.commit()
[2893332.432]  -> xdg_wm_base@9.get_xdg_surface(new id xdg_surface@25, wl_surface@20)
[2893332.445]  -> xdg_surface@25.get_toplevel(new id xdg_toplevel@26)
[2893332.452]  -> xdg_toplevel@26.set_title("Window Renderer")
[2893332.472]  -> wl_shm@5.create_pool(new id wl_shm_pool@27, fd 14, 4)
[2893332.486]  -> wl_shm_pool@27.create_buffer(new id wl_buffer@28, 0, 1, 1, 4, 0)
[2893332.508]  -> wl_shm_pool@27.destroy()
[2893332.512]  -> wl_compositor@4.create_surface(new id wl_surface@29)
[2893332.524]  -> wl_subcompositor@8.get_subsurface(new id wl_subsurface@30, wl_surface@29, wl_surface@20)
[2893332.534]  -> wl_subsurface@30.set_position(0, -24)
[2893332.542]  -> wp_viewporter@11.get_viewport(new id wp_viewport@31, wl_surface@29)
[2893332.550]  -> wp_viewport@31.set_destination(800, 24)
[2893332.557]  -> wl_surface@29.attach(wl_buffer@28, 0, 0)
[2893332.566]  -> wl_compositor@4.create_region(new id wl_region@32)
[2893332.572]  -> wl_region@32.add(0, 0, 800, 24)
[2893332.584]  -> wl_surface@29.set_opaque_region(wl_region@32)
[2893332.588]  -> wl_surface@29.commit()
[2893332.592]  -> wl_region@32.destroy()
[2893332.595]  -> wl_compositor@4.create_surface(new id wl_surface@33)
[2893332.601]  -> wl_subcompositor@8.get_subsurface(new id wl_subsurface@34, wl_surface@33, wl_surface@20)
[2893332.611]  -> wl_subsurface@34.set_position(-4, -24)
[2893332.618]  -> wp_viewporter@11.get_viewport(new id wp_viewport@35, wl_surface@33)
[2893332.626]  -> wp_viewport@35.set_destination(4, 624)
[2893332.633]  -> wl_surface@33.attach(wl_buffer@28, 0, 0)
[2893332.642]  -> wl_compositor@4.create_region(new id wl_region@36)
[2893332.647]  -> wl_region@36.add(0, 0, 4, 624)
[2893332.659]  -> wl_surface@33.set_opaque_region(wl_region@36)
[2893332.663]  -> wl_surface@33.commit()
[2893332.666]  -> wl_region@36.destroy()
[2893332.670]  -> wl_compositor@4.create_surface(new id wl_surface@37)
[2893332.675]  -> wl_subcompositor@8.get_subsurface(new id wl_subsurface@38, wl_surface@37, wl_surface@20)
[2893332.686]  -> wl_subsurface@38.set_position(800, -24)
[2893332.692]  -> wp_viewporter@11.get_viewport(new id wp_viewport@39, wl_surface@37)
[2893332.700]  -> wp_viewport@39.set_destination(4, 624)
[2893332.707]  -> wl_surface@37.attach(wl_buffer@28, 0, 0)
[2893332.716]  -> wl_compositor@4.create_region(new id wl_region@40)
[2893332.722]  -> wl_region@40.add(0, 0, 4, 624)
[2893332.733]  -> wl_surface@37.set_opaque_region(wl_region@40)
[2893332.737]  -> wl_surface@37.commit()
[2893332.740]  -> wl_region@40.destroy()
[2893332.744]  -> wl_compositor@4.create_surface(new id wl_surface@41)
[2893332.749]  -> wl_subcompositor@8.get_subsurface(new id wl_subsurface@42, wl_surface@41, wl_surface@20)
[2893332.760]  -> wl_subsurface@42.set_position(-4, 600)
[2893332.766]  -> wp_viewporter@11.get_viewport(new id wp_viewport@43, wl_surface@41)
[2893332.774]  -> wp_viewport@43.set_destination(808, 4)
[2893332.781]  -> wl_surface@41.attach(wl_buffer@28, 0, 0)
[2893332.790]  -> wl_compositor@4.create_region(new id wl_region@44)
[2893332.796]  -> wl_region@44.add(0, 0, 808, 4)
[2893332.807]  -> wl_surface@41.set_opaque_region(wl_region@44)
[2893332.811]  -> wl_surface@41.commit()
[2893332.814]  -> wl_region@44.destroy()
[2893332.818]  -> wl_surface@20.commit()
[2893332.821]  -> wl_display@1.sync(new id wl_callback@45)
[2893332.827] wl_keyboard@16.keymap(1, fd 7, 52124)
[2893338.492] wl_keyboard@16.repeat_info(33, 500)
[2893338.595] wl_display@1.error(xdg_wm_base@9, 4, "wl_surface@20 already has a buffer committed")
xdg_wm_base@9: error 4: wl_surface@20 already has a buffer committed
[2893338.817]  -> wl_buffer@24.destroy()
[2893339.257]  -> wl_surface@29.destroy()
[2893339.267]  -> wl_subsurface@30.destroy()
[2893339.270]  -> wp_viewport@31.destroy()
[2893339.274]  -> wl_surface@33.destroy()
[2893339.277]  -> wl_subsurface@34.destroy()
[2893339.280]  -> wp_viewport@35.destroy()
[2893339.284]  -> wl_surface@37.destroy()
[2893339.287]  -> wl_subsurface@38.destroy()
[2893339.290]  -> wp_viewport@39.destroy()
[2893339.293]  -> wl_surface@41.destroy()
[2893339.296]  -> wl_subsurface@42.destroy()
[2893339.299]  -> wp_viewport@43.destroy()
[2893339.302]  -> wl_buffer@28.destroy()
[2893339.306]  -> xdg_toplevel@26.destroy()
[2893339.309]  -> xdg_surface@25.destroy()
[2893339.313]  -> wl_surface@20.destroy()
[2893354.450]  -> wl_shm_pool@3.destroy()
[2893355.393]  -> wl_shm_pool@17.destroy()
[2893357.497]  -> wl_surface@18.destroy()
[2893357.522]  -> wl_subcompositor@8.destroy()
[2893357.536]  -> wp_viewporter@11.destroy()
[2893357.567]  -> xdg_wm_base@9.destroy()
[2893357.586]  -> zwp_relative_pointer_manager_v1@13.destroy()
[2893357.599]  -> zwp_pointer_constraints_v1@14.destroy()
```
</details>

### Here is the glfwinfo
<details>
<summary>glfwinfo</summary>

```
GLFW header version: 3.3.0
GLFW library version: 3.3.0
GLFW library version string: "3.3.0 Wayland EGL OSMesa clock_gettime evdev"
OpenGL context version string: "4.5 (Compatibility Profile) Mesa 19.0.4"
OpenGL context version parsed by GLFW: 4.5.0
OpenGL context flags (0x00000000):
OpenGL context flags parsed by GLFW:
OpenGL profile mask (0x00000002): compat
OpenGL profile mask parsed by GLFW: compat
OpenGL robustness strategy (0x00008261): none
OpenGL robustness strategy parsed by GLFW: none
OpenGL context renderer string: "AMD OLAND (DRM 2.50.0, 5.1.2-arch1-1-ARCH, LLVM 8.0.0)"
OpenGL context vendor string: "X.Org"
OpenGL context shading language version: "4.50"
OpenGL framebuffer:
 red: 8 green: 8 blue: 8 alpha: 8 depth: 24 stencil: 8
 samples: 0 sample buffers: 0
 accum red: 0 accum green: 0 accum blue: 0 accum alpha: 0 aux buffers: 0
Vulkan loader: available
Vulkan required instance extensions: missing
```
</details>

**Comment by costashatz** [comment]
@aleck099 did you manage to solve this issue?

**Comment by aleck099** [comment]
@costashatz Sorry, I don't have any solutions about this. I have given up using Wayland already.

**Comment by joeroback** [comment]
i too any encountering this on aarch64 + wayland + glfw 3.4 (ToT master)

**Comment by joeroback** [comment]
in order to get any of the examples to run, i had to go back to 3.2.1 .. but none of the examples on 3.2.1. have any window decorations

**Comment by elmindreda** [comment]
This issue should now be fixed with 094aa6d3c721397825cb1095c156353463ecafb5, which should be included in 3.3.7.
